src/main/java/io/cdap/plugin/servicenow/source/ServiceNowMultiSourceConfig.java

```diff
@@ -0,0 +1,140 @@
+ * the License.
+ */
+
+package io.cdap.plugin.servicenow.source;
+
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Macro;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.plugin.common.IdUtils;
+import io.cdap.plugin.servicenow.restapi.RestAPIResponse;
+import io.cdap.plugin.servicenow.source.apiclient.ServiceNowTableAPIClientImpl;
+import io.cdap.plugin.servicenow.source.apiclient.ServiceNowTableAPIRequestBuilder;
+import io.cdap.plugin.servicenow.source.util.ServiceNowConstants;
+import io.cdap.plugin.servicenow.source.util.Util;
+
+import org.apache.http.HttpStatus;
+import org.apache.oltu.oauth2.common.exception.OAuthProblemException;
+import org.apache.oltu.oauth2.common.exception.OAuthSystemException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Set;
+
+import javax.annotation.Nullable;
+
+import static io.cdap.plugin.servicenow.source.util.ServiceNowConstants.HEADER_NAME_TOTAL_COUNT;
+
+/**
+ * Configuration for the {@link ServiceNowMultiSource}.
+ */
+public class ServiceNowMultiSourceConfig extends ServiceNowBaseSourceConfig {
+  private static final Logger LOG = LoggerFactory.getLogger(ServiceNowMultiSourceConfig.class);
+
+  @Name(ServiceNowConstants.PROPERTY_TABLE_NAMES)
+  @Macro
+  @Description("The names of the ServiceNow table from which data to be fetched.")
+  private String tableNames;
+
+  /**
+   * Constructor for ServiceNowSourceConfig object.
+   *
+   * @param referenceName The reference name
+   * @param tableNameField The field name to hold the table name value
+   * @param clientId The Client Id for ServiceNow
+   * @param clientSecret The Client Secret for ServiceNow
+   * @param restApiEndpoint The rest API endpoint for ServiceNow
+   * @param user The user id for ServiceNow
+   * @param password The password for ServiceNow
+   * @param valueType The value type
+   * @param startDate The start date
+   * @param endDate The end date
+   */
+  public ServiceNowMultiSourceConfig(String referenceName, String tableNameField, String clientId,
+                                     String clientSecret, String restApiEndpoint, String user, String password,
+                                     String valueType, @Nullable String startDate, @Nullable String endDate,
+                                     String tableNames) {
+    super(referenceName, tableNameField, clientId, clientSecret, restApiEndpoint, user, password, valueType, startDate,
+      endDate);
+    this.tableNames = tableNames;
+  }
+
+  @Nullable
+  public String getTableNames() {
+    return tableNames;
+  }
+
+  /**
+   * Validates {@link ServiceNowMultiSourceConfig} instance.
+   */
+  public void validate(FailureCollector collector) {
+    // Validates the given referenceName to consists of characters allowed to represent a dataset.
+    IdUtils.validateReferenceName(referenceName, collector);
+
```

Comment:
you can call super.validate() + add multisource specific validations here

Reply:
Refactored the code. Called `super.validate()` and added validation for properties used in Multi Source 7661e08